### PR TITLE
Adds lgtm changes and addresses core dev comments

### PR DESCRIPTION
**Pull request by obrien951**
## Checklist
- [ ] Tests added for any new features (no new features)
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests) (tests that run before changes also run after changes)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by JonathonMisiewicz** [comment]
How does this PR relate to the very similar #1635? Do we need both?

**Comment by obrien951** [comment]
@JonathonMisiewicz It's a replacement of #1635 . I wanted to be able to test some of the fixes before I made them, and I couldn't remember which conda environment to compile that snapshot of psi with. Closing the old one presently.

**Review by loriab — APPROVED** [review]
lgtm! thanks for refreshing the PR

File: psi4/src/psi4/libfock/cubature.cc [hunk 1]
```diff
@@ -4565,8 +4565,7 @@ void NaiveGridBlocker::block() {
 
     ::memcpy((void *)x_, (void *)x_ref_, sizeof(double) * npoints_);
     ::memcpy((void *)y_, (void *)y_ref_, sizeof(double) * npoints_);
-    ::memcpy((void *)z_, (void *)z_ref_, sizeof(double) * npoints_);
-    ::memcpy((void *)w_, (void *)w_ref_, sizeof(double) * npoints_);
+    ::memcpy((void *)z_, (void *)z_ref_, sizeof(double) * npoints_); ::memcpy((void *)w_, (void *)w_ref_, sizeof(double) * npoints_);
```
[the remark this answers is not in the document]
Comment (obrien951) [reply psi4/src/psi4/libfock/cubature.cc:4568]: definite slip

**Review by robertodr — APPROVED** [review]

**Review by jturney — APPROVED** [review]

**Pull request merged**